apps/examples/fs_helper/fs_rw_test.c

```diff
@@ -0,0 +1,168 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/time.h>
+
+/****************************************************************************
+ * Definitions
+ ****************************************************************************/
+#define TEST_FILE_PATH "/mnt/"
+#define TEST_FILE_NAME "test"
+#define USAGE																\
+	"\n Filesystem R/W Test\n"												\
+	"\n usage: fs_rw_test [Size] [Iteration] [Operation] [Log]\n"		\
+	"\n [Size] : Data size (byte)\n"										\
+	" [Iteration] : Number of repetitions\n"								\
+	" [Operation] : 1.Read/Write / 2.Read / 3.Write\n"						\
+	" [Log] : Log Print Option [y/n]\n"										\
+	"\n ex) fs_rw_test 128 30 1 y\n"
+
+/****************************************************************************
+ * fs_rw_test
+ ****************************************************************************/
+
+static inline long get_time(void)
+{
+	struct timeval tv;
+	gettimeofday(&tv, NULL);
+	return (tv.tv_sec * 1000 + tv.tv_usec / 1000);
+}
+
+static void rw_testlog(char *printopt, const char *format, ...)
+{
+	va_list ap;
+	if (printopt[0] == 'y' || printopt[0] == 'Y') {
+		va_start(ap, format);
+		vfprintf(stdout, format, ap);
+		va_end(ap);
+	}
+}
+
+static FILE *t_fopen(const char *path, const char *mode, char *printopt)
+{
+	FILE *fp;
+	long t, dt;
+	t = get_time();	
+	fp = fopen(path, mode);
+	if (fp == NULL) {
+		printf("fopen Failed\n");
+		return NULL;
+	}
+	dt = get_time() - t;
+	rw_testlog(printopt, "fopen(%c):%dms, ", mode[0], dt);
+	return fp;	
+}
+
+static void t_fwrite(const void *data, size_t size, size_t n_items, FILE *stream, char *printopt)
```

Comment:
streaming is more slower than posix  type.
why don't you add functions to test posix type ? =)

Reply:
Do you think is it enough answer???????? Is there any problem to add a test of posix type?

Reply:
modify.

fopen, fwrite, and fread are used only in certain cases, but open, write, and read are used in all cases.